src/sources/file.rs

```diff
@@ -782,14 +807,30 @@ mod tests {
     fn file_create_event() {
         let line = Bytes::from("hello world");
         let file = "some_file.rs";
+
+        // Internal offset is the end of the line including the newline char,
+        // whereas the `line` variable contains the line with the newline stripped.
+        // So 11 characters in `line` -> internal offset is 12 including the newline.
```

Comment:
Do you have any thoughts on accounting for Windows, where the end-of-line sequence is a two-byte pair of characters?

Reply:
Ahh yes, I did not consider Windows' line endings. I'll think about this a little bit, presumably there's a point before the endings are stripped where we can tell.

If I can't find a good way we can fall back to just using the end-of-line offset instead, which is fine for the ordering purpose.

Reply:
Getting the beginning of line offset accurately might require plumbing it into the file server code in `lib/file-server/src`, possibly returning it with the bytes in `FileWatcher::read_line`.

Reply:
While looking into this, I noticed that we already have `line_delimiter_as_bytes` in scope, and hence can use the length of that in the calculation instead of my initial fixed `-1`. 

I still agree though that getting this out of `FileWatcher::read_line` instead of trying to re-calculate it here is the more correct approach. The value is definitely available there, it's just a matter of making it available here as well. I'm still looking at how to implement this, but as a relative newcomer to rust there's a bit of a learning curve for me here. Up to you I guess whether you consider the implemented approach good enough for now, or would prefer to wait for the `FileWatcher::read_line` approach.